tests/flow/test_entity_update.py

```diff
@@ -113,9 +113,30 @@ def test11_property_map_from_parameter(self):
         self.env.assertEqual(result.result_set, expected_result)
 
     # Fail update an entity property when left hand side is not alias
-    def test12_fail_update_property_of_non_alias_enetity(self):
+    def test12_fail_update_property_of_non_alias_entity(self):
         try:
             graph.query("MATCH P=() SET nodes(P).prop = 1 RETURN nodes(P)")
             self.env.assertTrue(False)
         except ResponseError as e:
             self.env.assertContains("RedisGraph does not currently support non-alias references on the left-hand side of SET expressions", str(e))
+
+    # Fail when a property is a complex type nested within an array type
+    def test13_invalid_complex_type_in_array(self):
+        # Skip this test if running under Valgrind, as it causes a memory leak.
+        if Env().envRunner.debugger is not None:
+            Env().skip()
+
+        # Test setting an array property containing a node
+        try:
+            graph.query("MATCH (a) SET a.v = [a]")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Property values can only be of primitive types or arrays of primitive types", str(e))
+
+        # Test updating all properties to a map with a
+        # nested array property containing NULL
+        try:
+            graph.query("MATCH (a) SET a = {v: ['str', [1, NULL]]}")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Property values can only be of primitive types or arrays of primitive types", str(e))
```

Comment:
We should also test for all cases where an array contains a restricted type:
1. map
2. node
3. edge
4. null